WooCommerce/Classes/ViewRelated/Orders/Order Details/Address Edit/EditAddressFormViewModel.swift

```diff
@@ -93,12 +120,28 @@ final class EditAddressFormViewModel: ObservableObject {
     ///
     func updateRemoteAddress(onFinish: @escaping (Bool) -> Void) {
         let updatedAddress = fields.toAddress(selectedCountry: selectedCountry)
-        let modifiedOrder = order.copy(shippingAddress: updatedAddress)
-        let action = OrderAction.updateOrder(siteID: order.siteID, order: modifiedOrder, fields: [.shippingAddress]) { [weak self] result in
+
+        let modifiedOrder: Yosemite.Order
+        switch type {
+        case .shipping:
+            modifiedOrder = order.copy(shippingAddress: updatedAddress)
+        case .billing:
+            modifiedOrder = order.copy(billingAddress: updatedAddress)
+        }
+
+        let action = OrderAction.updateOrder(siteID: order.siteID,
+                                             order: modifiedOrder,
+                                             fields: [type == .shipping ? .shippingAddress : .billingAddress]) { [weak self] result in
```

Comment:
This comparison worries me because it will break if we add another address type, but that is very unlikely to happen 😅

Reply:
That’s exactly why in 2 cases above I’ve used switch-case without default handling 😄
Tried here too, but field type is internal to Networking.